Test with playwright

naxoc opened the pull request:
Closes #129

## Proposed changes

This adds a Playwright e2e setup so we can better test cross-browser issues and interactions that are too complicated for jsdom.

## To test

1. `yarn e2e` (should work "cold" and with Storybook already running)

✅ CI tests should pass.

naxoc committed:
Start using Playwright

naxoc committed:
Try stuff

naxoc committed:
Fix typo

mirka committed:
Lint config

msurdi commented on .github/workflows/nodejs.yml:
```diff
@@ -38,3 +38,13 @@ jobs:
+            run: npx playwright install --with-deps
+            - name: Serve Storybook and run tests
+            run: |
+                yarn storybbok \
```
Are we 100% sure we want to go our own way on writing, running and debugging tests vs using the [official test tools setup](https://storybook.js.org/addons/@storybook/test-runner) by storybook? I'm personally not that convinced, but happy to do it this way too if the rest of the team thinks it is required, easier or just enough for our use case.

naxoc replied:
I all for using what is easier, so let's use the official tools if you think that is easier!

msurdi replied:
I think having the benefits of community support and tooling are worth to give it a try. Disclaimer: I've never used it myself, just seen this in use by a sibling team at a previous job and didn't hear many complaints.

[Here's an example](https://mealdrop.vercel.app/storybook/?path=/story/userflows-app--end-to-end) I've just found of some of the tooling you get for writing/debugging tests, if you want to give it a try and see how it feels. And [here](https://github.com/yannbf/mealdrop/tree/main/src/pages) seems to be the source code of how the tests are organized.

One thing I like is that the test code sticks together with the "story", so it is easier to keep together a component's documentation, tests, etc.

*WARNING: That website might make you hungry*

mirka replied:
Playwright tests for iso-editor are going to be more focused on functionality rather than UI/components for now (13901df), but I'm hoping to go the `@storybook/test-runner` route for wp-components! 🔥

mirka committed:
Tweak test matchers

I think it makes sense to keep e2e test files closer to the implementation files (along with unit tests), rather than group them with the Storybook files. This is because I think we'll be testing different functionality (undo, collab, etc.) using some of the same stories.

naxoc committed:
Use role selector

mirka committed:
Add screenshots/video

naxoc committed:
Move file

mirka committed:
Add experimental features flag

# Conflicts:
#	package.json

mirka committed:
Fixup eslint error

naxoc committed:
Fix borked indent in yaml

naxoc committed:
Use run

mirka committed:
Integrate webserver

# Conflicts:
#	.github/workflows/nodejs.yml

mirka committed:
Add iso-editor default e2e

naxoc committed:
Use own action file

mirka committed:
Exclude e2e tests from jest

naxoc committed:
Merge branch 'test/with-playwright' of github.com:Automattic/isolated-block-editor into test/with-playwright

naxoc committed:
Merge branch 'test/with-playwright' of github.com:Automattic/isolated-block-editor into test/with-playwright

naxoc committed:
Also on push

naxoc committed:
Woops

naxoc committed:
Move

naxoc committed:
Fix path

naxoc committed:
Move again

naxoc committed:
Now then?

naxoc committed:
Just use one actions file

mirka committed:
Disable parallelism for now

Can't get it to work! Fails on the Firefox test only. Only fails on cold starts (`yarn e2e` without the storybook server previously running).

naxoc committed:
Now?

naxoc committed:
Merge branch 'test/with-playwright' of github.com:Automattic/isolated-block-editor into test/with-playwright

naxoc committed:
Move test file

mirka committed:
Wrap toolbar test in description
